cs/ccxt/base/Throttler.cs

```diff
@@ -73,23 +89,85 @@ private async Task loop()
 
     }
 
-    public async Task<Task> throttle(object cost2)
+    private async Task rollingWindowLoop()
     {
+        while (this.running)
+        {
+            lock (queueLock)
+            {
+                if (this.queue.Count == 0)
+                {
+                    this.running = false;
+                    continue;
+                }
+            }
+            (Task, double) first;
+            lock (queueLock)
+            {
+                first = this.queue.Peek();
+            }
+            var task = first.Item1;
+            var cost = first.Item2;
+            var now = milliseconds();
+            timestamps = timestamps.Where(t => now - t.timestamp < Convert.ToDouble(this.config["windowSize"])).ToList();
+            var totalCost = timestamps.Sum(t => t.cost);
+            if (totalCost + cost <= Convert.ToDouble(this.config["maxWeight"]))
```

Comment:
@carlosmiei can't we just ```suggestion
            if (totalCost + cost <=  (double)(this.config["maxWeight"]))
```

Reply:
and also, same two lines above